### DISCUSSION TITLE
pos def ?

### DISCUSSION THREAD

**Discussion by microprediction** [skfolio/skfolio#28] [Ideas]
I don't have a minimal example but it was relayed to me that there are a few times when cov estimation might benefit from a little post-estimation scrubbing (e.g. nearest pos def). By all means steal from the utilities here. 

https://github.com/microprediction/precise/tree/main/precise/skaters/covarianceutil

**Comment by HugoDelatte** [comment]
Thanks for the feedback and for the `microprediction` utilities!

Covariance estimators have already a parameter named [nearest](https://github.com/skfolio/skfolio/blob/51a91b6a70bf28f4207e0333e44ad48f36a822a0/src/skfolio/moments/covariance/_base.py#L25) to handle that case.

However, the default is set to `False`, so I wonder if we should change it to `True`?

The implementation details are [here](https://github.com/skfolio/skfolio/blob/51a91b6a70bf28f4207e0333e44ad48f36a822a0/src/skfolio/utils/stats.py#L304C5-L304C16). I'll compare it with your [implementation](https://github.com/microprediction/precise/blob/1c62866579b122e04c746174c722316f49bbaf23/precise/skaters/covarianceutil/covfunctions.py#L167C5-L167C20).

One difference is that I also test for Cholesky decomposition because I noticed that the Cholesky decomposition performed by numpy can sometimes fail even when the matrix is PD due to floating-point inaccuracies (specially on large matrices) as seen [in this test](https://github.com/skfolio/skfolio/blob/51a91b6a70bf28f4207e0333e44ad48f36a822a0/tests/test_utils/test_stats.py#L91) (and inversely, numpy can return a decomposition on a matrix which is not PD).

**Comment by microprediction** [comment]
I missed that param, sorry!
